lib/tasks/server/express-server.js

```diff
@@ -116,7 +116,7 @@ class ExpressServerTask extends Task {
   }
 
   processAppMiddlewares(options) {
-    if (this.project.has(this.serverRoot)) {
+    if (this.project.has([this.serverRoot, 'index'].join(path.sep))) {
```

Comment:
has logic: https://github.com/ember-cli/ember-cli/blob/master/lib/models/project.js#L342

Reply:
This would break folks using `server.js` (which is also supported).